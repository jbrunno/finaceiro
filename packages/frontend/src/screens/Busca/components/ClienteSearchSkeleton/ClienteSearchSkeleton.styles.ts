import { Box, Skeleton } from '@frontend/components';
import { css, styled } from '@frontend/styles';

export const CardList = styled('div')(
  ({ theme }) => css`
    padding: 0;
    display: grid;
    row-gap: ${theme.fn.spacing(16)};
    padding: ${theme.fn.spacing(0, 8)};
    grid-auto-rows: min-content;
    &:after {
      content: '';
      height: 1px;
    }
  `,
);

export const StyledBox = styled(Box)`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledSkeleton = styled(Skeleton)`
  margin-bottom: ${({ theme }) => theme.fn.spacing(8)};
`;
