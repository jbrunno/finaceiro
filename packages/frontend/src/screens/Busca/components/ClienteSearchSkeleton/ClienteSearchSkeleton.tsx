import React from 'react';
import { Card, CardContent, CardHeader, Skeleton } from '@frontend/components';
import {
  CardList,
  StyledBox,
  StyledSkeleton,
} from './ClienteSearchSkeleton.styles';

interface ClienteSearchSkeletonProps {
  cards: number;
}

export function ClienteSearchSkeleton({ cards }: ClienteSearchSkeletonProps) {
  const mapArr = Array(cards).fill('');
  return (
    <CardList>
      {mapArr.map((el) => {
        return (
          <Card key={`${el}-${Math.random()}`}>
            <CardHeader
              subheader={
                <Skeleton
                  variant="rectangular"
                  height={30}
                  width="100%"
                  animation="wave"
                />
              }
            />
            <CardContent>
              <StyledBox>
                <StyledSkeleton
                  variant="rectangular"
                  height={35}
                  width="100%"
                  animation="wave"
                />
                <StyledSkeleton
                  variant="rectangular"
                  height={35}
                  width="100%"
                  animation="wave"
                />
              </StyledBox>
            </CardContent>
          </Card>
        );
      })}
    </CardList>
  );
}
