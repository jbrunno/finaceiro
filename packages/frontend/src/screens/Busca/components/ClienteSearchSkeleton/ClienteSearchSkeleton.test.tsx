import React from 'react';
import { render } from '@test-utils/component-testing';
import { ClienteSearchSkeleton } from './ClienteSearchSkeleton';

test('renders the CardCustomer', () => {
  const { container } = render(<ClienteSearchSkeleton cards={3} />);
  expect(container).toMatchSnapshot();
});
