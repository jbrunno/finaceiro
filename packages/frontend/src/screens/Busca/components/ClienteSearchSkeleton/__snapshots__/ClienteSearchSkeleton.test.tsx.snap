// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CardCustomer 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  50% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }

  100% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  50% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }

  100% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  padding: 0;
  display: grid;
  row-gap: 1rem;
  padding: 0rem 0.5rem;
  grid-auto-rows: min-content;
}

.emotion-0:after {
  content: '';
  height: 1px;
}

.emotion-3 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  overflow: visible;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px -1px rgba(233, 233, 233, 0.8),0px 4px 5px rgba(233, 233, 233, 0.14),0px 1px 10px rgba(233, 233, 233, 0.12);
  background-color: #fff;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-6 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B4199;
  display: block;
}

.emotion-7 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: 1.2em;
  position: relative;
  overflow: hidden;
  -webkit-mask-image: -webkit-radial-gradient(white, black);
}

.emotion-7::after {
  -webkit-animation: animation-0 1.6s linear 0.5s infinite;
  animation: animation-0 1.6s linear 0.5s infinite;
  background: linear-gradient(
            90deg,
            transparent,
            #0000000A,
            transparent
          );
  content: '';
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
}

.emotion-8 {
  padding: 16px;
}

.emotion-8:last-child {
  padding-bottom: 24px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-12 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: 1.2em;
  position: relative;
  overflow: hidden;
  -webkit-mask-image: -webkit-radial-gradient(white, black);
  margin-bottom: 0.5rem;
}

.emotion-12::after {
  -webkit-animation: animation-0 1.6s linear 0.5s infinite;
  animation: animation-0 1.6s linear 0.5s infinite;
  background: linear-gradient(
            90deg,
            transparent,
            #0000000A,
            transparent
          );
  content: '';
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2 emotion-3"
    >
      <div
        class="MuiCardHeader-root emotion-4"
      >
        <div
          class="MuiCardHeader-content emotion-5"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader emotion-6"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-7"
              style="width: 100%; height: 30px;"
            />
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root emotion-8"
      >
        <div
          class="emotion-9 MuiBox-root emotion-10"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2 emotion-3"
    >
      <div
        class="MuiCardHeader-root emotion-4"
      >
        <div
          class="MuiCardHeader-content emotion-5"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader emotion-6"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-7"
              style="width: 100%; height: 30px;"
            />
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root emotion-8"
      >
        <div
          class="emotion-9 MuiBox-root emotion-10"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2 emotion-3"
    >
      <div
        class="MuiCardHeader-root emotion-4"
      >
        <div
          class="MuiCardHeader-content emotion-5"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader emotion-6"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-7"
              style="width: 100%; height: 30px;"
            />
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root emotion-8"
      >
        <div
          class="emotion-9 MuiBox-root emotion-10"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-11 emotion-12"
            style="width: 100%; height: 35px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
